### ISSUE TITLE
Switch from Bors to GitHub Merge Queues

### ISSUE THREAD

**Issue by metasim** [georust/gdal#452]
Bors is going away:

![image](https://github.com/georust/gdal/assets/131013/e44aaaf6-a7b7-48a8-8fe7-0fec4fe3defd)

The `geo` team has converted their repo to use GitHub Merge Queues. Seems like following their lead and not having to host our own bors infrastructure is the easiest path forward.

Per [this discussion](https://discord.com/channels/598002550221963289/598002550221963291/1159148505433128990), we can use the following as references for the migration:

* https://github.com/georust/geo/pull/1037
* https://github.com/georust/proj/pull/179#issuecomment-1684521823

We will need to determine the process impacts, i.e. what do we need to change in the way we review, accept, and merge PRs.

**Comment by michaelkirk** [comment]
Some notes from the geo transition to merge queues (which seem to be working fine BTW).

They are my own notes and might not be 100% correct:

Within repo > settings > branches

You need a branch protection rule for the main branch.

Then you need to apply at least these settings:

## status checks
<img width="396" alt="Screenshot 2023-10-04 at 10 22 31" src="https://github.com/georust/gdal/assets/217057/a85c62ad-c78c-419a-97d3-664b1cb160b6">

You need to enter every status check and keep this list in sync as you add/remove jobs, just like you have been with your bors.toml. Note that the geo repo only lists one status check because we have an [aggregate job](https://github.com/georust/geo/blob/main/.github/workflows/test.yml#L19) that combines all the other jobs.

## enable merge queue

<img width="401" alt="Screenshot 2023-10-04 at 10 22 20" src="https://github.com/georust/gdal/assets/217057/fcf23ccd-e0d0-4426-9f67-3cf7a2af8fb4">

## add `merge_group` to your workflow triggers (inside the `on` key)
```diff
 on:
   push:
     branches:
       - main
       - staging
       - trying
       - release/**
   pull_request:
+  merge_group:
```
e.g. in [geo](https://github.com/georust/geo/blob/main/.github/workflows/test.yml#L9)

TBH I found the transition surprisingly complicated, but it was mostly due to an arbitrary detail in how we implemented our aggregate `ci-status` job.

The short version is: A GH status can pass, fail, or be skipped. Our aggregate job would be skipped if any of the sub-jobs failed. I actually never realized this and assumed it would fail. But it didn't matter because bors considered a skip as a failure. GH merge-queues did the opposite - it would proceed when it's status check was "skipped". 

Ultimately it didn't require a big change, but took me an annoyingly long time to figure out what the issue was. And since it doesn't look like you have an aggregate job it shouldn't affect you anyway.

Good luck!

**Comment by metasim** [comment]
>  bors considered a skip as a failure. GH merge-queues did the opposite - it would proceed when it's status check was "skipped".

I'm willing to be this insight alone will save one of us days of insanity! 😅
